From efe8956fd704ba6c0a04da457e78fd3e1e21dbf8 Mon Sep 17 00:00:00 2001
From: Bryan Brattlof <bb@ti.com>
Date: Mon, 10 Feb 2025 22:03:41 -0600
Subject: [PATCH 66/68] PENDING: arm64: dts: ti: k3-am62l: move sdhci to use
 the SCMI

Now that our SCMI is working, let's remove the fixed clock and use the
proper smci phandles for the MMC interface

Signed-off-by: Bryan Brattlof <bb@ti.com>
---
 arch/arm64/boot/dts/ti/k3-am62l-main.dtsi |  5 +++++
 arch/arm64/boot/dts/ti/k3-am62l3-evm.dts  | 14 --------------
 2 files changed, 5 insertions(+), 14 deletions(-)

diff --git a/arch/arm64/boot/dts/ti/k3-am62l-main.dtsi b/arch/arm64/boot/dts/ti/k3-am62l-main.dtsi
index 9eca95eae04b..28a6f924e461 100644
--- a/arch/arm64/boot/dts/ti/k3-am62l-main.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-am62l-main.dtsi
@@ -47,6 +47,11 @@ sdhci1: mmc@fa00000 {
 		reg = <0x00 0x0fa00000 0x00 0x1000>,
 		      <0x00 0x0fa08000 0x00 0x400>;
 		interrupts = <GIC_SPI 237 IRQ_TYPE_LEVEL_HIGH>;
+		power-domains = <&scmi_pds 26>;
+		clocks = <&scmi_clk 106>, <&scmi_clk 107>;
+		clock-names = "clk_ahb", "clk_xin";
+		assigned-clocks = <&scmi_clk 107>;
+		assigned-clock-parents = <&scmi_clk 0>;
 		bus-width = <4>;
 		ti,clkbuf-sel = <0x7>;
 		ti,otap-del-sel-legacy = <0x0>;
diff --git a/arch/arm64/boot/dts/ti/k3-am62l3-evm.dts b/arch/arm64/boot/dts/ti/k3-am62l3-evm.dts
index 78a636fdaaa9..98ffec498195 100644
--- a/arch/arm64/boot/dts/ti/k3-am62l3-evm.dts
+++ b/arch/arm64/boot/dts/ti/k3-am62l3-evm.dts
@@ -23,18 +23,6 @@ memory@80000000 {
 		device_type = "memory";
 		bootph-all;
 	};
-
-	/*
-	 * Boot ROM will have MMC configure for us when booting via SD
-	 * card. Put this here so the MMC driver doesn't try to change
-	 * anything for us and break boot.
-	 */
-	 clk_xin: clock {
-		 compatible = "fixed-clock";
-		 clock-frequency = <5000000>;
-		 #clock-cells = <0>;
-		 bootph-all;
-	 };
 };
 
 &main_uart0 {
@@ -68,8 +56,6 @@ AM62LX_IOPAD(0x01b8, PIN_OUTPUT, 0)	/* (C13) UART0_TXD */
 };
 
 &sdhci1 {
-	clocks = <&clk_xin>;
-	clock-names = "clk_xin";
 	pinctrl-names = "default";
 	pinctrl-0 = <&main_mmc1_pins_default>;
 	status = "okay";
-- 
2.47.2

